controllers/storageconsumer/storageconsumer_controller.go

```diff
@@ -52,11 +84,439 @@ func (r *StorageConsumerReconciler) Reconcile(ctx context.Context, request recon
+
+	r.cephClientHealthChecker = &rookCephv1.CephClient{}
+	r.cephClientHealthChecker.Name = fmt.Sprintf("%s-%s", r.storageConsumer.Name, cephClientHealthCheckerSuffix)
+	r.cephClientHealthChecker.Namespace = r.namespace
+
+}
+
+func (r *StorageConsumerReconciler) reconcilePhases() (reconcile.Result, error) {
+	if r.storageConsumer.GetDeletionTimestamp().IsZero() {
+		if !contains(r.storageConsumer.GetFinalizers(), storageConsumerFinalizer) {
+			r.Log.Info("Finalizer not found for StorageConsumer. Adding finalizer.", "StorageConsumer", klog.KRef(r.storageConsumer.Namespace, r.storageConsumer.Name))
+			r.storageConsumer.ObjectMeta.Finalizers = append(r.storageConsumer.ObjectMeta.Finalizers, storageConsumerFinalizer)
+			if err := r.Client.Update(r.ctx, r.storageConsumer); err != nil {
+				r.Log.Info("Failed to update StorageConsumer with finalizer.", "StorageConsumer", klog.KRef(r.storageConsumer.Namespace, r.storageConsumer.Name))
+				return reconcile.Result{}, err
+			}
+		}
+	}
+
+	r.storageConsumer.Status.State = v1alpha1.StorageConsumerStateConfiguring
+
+	if err := r.reconcileCephBlockPool(); err != nil {
+		return reconcile.Result{}, err
+	}
+
+	if err := r.reconcileCephFilesystemSubVolumeGroup(); err != nil {
+		return reconcile.Result{}, err
+	}
+
+	if err := r.reconcileCephClientRBDProvisioner(); err != nil {
+		return reconcile.Result{}, err
+	}
+
+	if err := r.reconcileCephClientRBDNode(); err != nil {
+		return reconcile.Result{}, err
+	}
+
+	if err := r.reconcileCephClientCephFSProvisioner(); err != nil {
+		return reconcile.Result{}, err
+	}
+
+	if err := r.reconcileCephClientCephFSNode(); err != nil {
+		return reconcile.Result{}, err
+	}
+
+	if err := r.reconcileCephClientHealthChecker(); err != nil {
+		return reconcile.Result{}, err
+	}
+
+	if !r.areRookCephResourcesReady() {
+		return reconcile.Result{Requeue: true, RequeueAfter: 5 * time.Second}, nil
+	}
+
+	r.storageConsumer.Status.State = v1alpha1.StorageConsumerStateReady
+	r.storageConsumer.Status.GrantedCapacity = r.storageConsumer.Spec.Capacity
 
 	return reconcile.Result{}, nil
 }
 
+func (r *StorageConsumerReconciler) reconcileCephBlockPool() error {
+
+	storageClusterList := ocsv1.StorageClusterList{}
+	err := r.Client.List(r.ctx, &storageClusterList, client.InNamespace(r.namespace))
+	if err != nil {
+		return fmt.Errorf("Error fetching StorageClusterList. %+v", err)
+	}
+
+	var failureDomain string
+	if len(storageClusterList.Items) == 1 {
+		failureDomain = storageClusterList.Items[0].Status.FailureDomain
+	}
+
+	capacity := r.storageConsumer.Spec.Capacity.String()
+	desired := &rookCephv1.CephBlockPool{
+		Spec: rookCephv1.NamedBlockPoolSpec{
+			PoolSpec: rookCephv1.PoolSpec{
+				FailureDomain: failureDomain,
+				Replicated: rookCephv1.ReplicatedSpec{
+					Size:                     3,
+					ReplicasPerFailureDomain: 1,
+					TargetSizeRatio:          .49,
+				},
+				EnableRBDStats: false,
+				Quotas: rookCephv1.QuotaSpec{
+					MaxSize: &capacity,
```

Comment:
Does `capacity` parses this regex `^[0-9]+[\.]?[0-9]*([KMGTPE]i|[kMGTPE])?$`?

Reply:
capacity is derived from `resource.Quantity` , so the parsing is taken care of it